### DISCUSSION TITLE
listen to change in [lightbox]

### DISCUSSION THREAD

**Discussion by rokozeit** [MurhafSousli/ngx-gallery#549] [Q&A] [answered]
Hello,

first of all, I am quite new to Angular. Thanks for the great work.

My question is, how can I listen to `(indexChange)`, `(itemsChange)`, `(itemClick)`,`(thumbClick)` from a [lightbox]? I followed the instructions for lightbox on https://ngx-gallery.netlify.app/#/lightbox. But I could not figure out how to listen to changes can clicks.

These are my code snippets

**HTML-Template:**

```
  <mat-grid-tile colspan="1" *ngFor="let item of mediaItems; let i = index" [lightbox]="i" [gallery]="galleryId">
  ... some items ...
  </mat-grid-tile>
```

**TS:**

```
  galleryRef: GalleryRef;
  mediaItems: GaleryItem[] = [];
  galleryId = 'myGallery';

  constructor(private gallery: Gallery) {
    this.galleryRef = this.gallery.ref(this.galleryId);
    this.galleryRef.setConfig({dots: true, thumbAutosize: true})
  }
  ngOnInit() {
    this.mediaItems = [{...},{...},...]
    this.galleryRef.load(items)
  }
```

**Comment by MurhafSousli** [comment, marked as answer]
Hi @rokozeit, You can access those streams from the `galleryRef`, see the API https://github.com/MurhafSousli/ngx-gallery/wiki/Gallery-API#galleryref-api

```ts
this.galleryRef.itemClick.subscribe(x => {
  // do something
})
```

**Reply by rokozeit** [reply]
Wow, you are fast. Thank you very much. It works.
